### PR TITLE
Added encryption, decryption of strings and updated package of userun…

### PR DESCRIPTION
**Pull request by humanjas**
…ittest
#21

# Introduction
Added the encryption and decryption of strings using the AES-CBC protocol. Next to that, I updated the min sdk to 19 and fixed a package bug in userunittest and conversioncontrollerunittest.

# Purpose
The encryption and decryption of personal information, so it is not directly readable from the database.

# Reason
This way, even if the database gets compromised, the data will not directly be readable.

# Future Applicability
It can be used in the future for other crucial personal information attributes.

# PR Reflection
This method uses the AES-CBC encryption protocol. So comprehension about this protocol is a pre.

### CODE REVIEW

File: app/src/main/java/nl/tudelft/b_b_w/controller/CryptoController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package nl.tudelft.b_b_w.controller;
+
+
+import org.encryptor4j.util.TextEncryptor;
+
+import java.security.GeneralSecurityException;
+
+/**
+ * Created by jasper on 22/05/2017.
```
Comment (ashay123) [inline app/src/main/java/nl/tudelft/b_b_w/controller/CryptoController.java]: Please add to the comment what the class actually does.

File: app/src/main/java/nl/tudelft/b_b_w/controller/CryptoController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+public class CryptoController {
+
+    /**
+     * Class attributes
+     * @param SECRET_KEY The key of the encryption
+     * @param textEncryptor The encryptor which uses AES Encryption in CBC mode with a maximum
+     *                      permitted key length of 256bit.
+     */
+    private final String SECRET_KEY = "k7JKPUDoGhTSo4by";
```
Comment (ashay123) [inline app/src/main/java/nl/tudelft/b_b_w/controller/CryptoController.java]: Why is the `SECRET_KEY` hardcoded?

File: app/src/main/java/nl/tudelft/b_b_w/controller/CryptoController.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * encryptString method
+     * @param data given data
+     * @return encryption of data
+     * @throws GeneralSecurityException if string is larger than 256 bit
+     */
+    public String encryptString(String data) {
+        try {
+            return textEncryptor.encrypt(data);
+        }catch (GeneralSecurityException e){
```
Comment (ashay123) [inline app/src/main/java/nl/tudelft/b_b_w/controller/CryptoController.java]: You missed a space before `catch`.

File: app/src/main/java/nl/tudelft/b_b_w/controller/CryptoController.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * decryptString method
+     * @param data given encrypted data
+     * @return decrypted data
+     * @throws GeneralSecurityException if string is larger than 256 bit
+     */
+    public String decryptString(String data) {
+        try {
+            return textEncryptor.decrypt(data);
+        }catch (GeneralSecurityException e){
```
Comment (ashay123) [inline app/src/main/java/nl/tudelft/b_b_w/controller/CryptoController.java]: Again, space.

File: app/src/main/java/nl/tudelft/b_b_w/controller/CryptoController.java (hunk 2 above)
Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/controller/CryptoController.java]: Whether you hardcode it, or generate it, it will result in the same.

Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/controller/CryptoController.java]: It is better to read it in from a file later though

**Comment by humanjas** [comment]
Updated checkstyle according to comments from ashay

**Comment by humanjas** [comment]
Since we are using test driven development, it is mandatory that you write your tests prior to writing the functionality. So it is not needed to write about tests of your functionality in the pull request.

**Comment by nzarin** [comment]
Can you explain why you updated the SDK to 19? It's a bit unclear to me

**Comment by humanjas** [comment]
To use Base64 encoding/decoding of data, you need to have a minimum sdk of 19

**Pull request closed without merging**